src/System.Reflection.Emit/tests/FieldBuilder/FieldBuilderSetConstant.cs

```
@@ -0,0 +1,353 @@
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_DateTime", typeof(DateTime), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Char", typeof(Char), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_String", typeof(string), FieldAttributes.Public);
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_FBTestEnum", typeof(FBTestEnum), FieldAttributes.Public);
+            VerificationHelper(field, null, typeof(ArgumentException));
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Decimal", typeof(Decimal), FieldAttributes.Public);
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+
+            field = TypeBuilder.DefineField("Field_NegTest2_Object", typeof(Object), FieldAttributes.Public);
+            VerificationHelper(field, new object(), typeof(ArgumentException));
+        }
+
+        [Fact]
+        public void NegTest3()
+        {
+            try
+            {
+                FieldBuilder field = TypeBuilder.DefineField("Field_NegTest3", typeof(FBTestStruct), FieldAttributes.Public);
+                VerificationHelper(field, null, typeof(ArgumentException));
+            }
+            catch (InvalidOperationException)
```

Why we eat this exception?